### PR TITLE
IOW-283 - routing

### PR DESCRIPTION
**Pull request by dsteinich**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,9 @@
...
-*.nar
-*.ear
-*.zip
-*.tar.gz
-*.rar
-
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
-hs_err_pid*
+/.idea/
+/aqts-capture-dvstat-transform.iml
```
Comment (ssoper-usgs) [inline .gitignore]: might want to add `aqts-capture-stattype-router.iml` on this line

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+            <id>wma-maven-snapshots</id>
+            <name>wma-maven-snapshots</name>
+            <url>${wma.maven.url}/wma-maven-snapshots</url>
+        </snapshotRepository>
+    </distributionManagement>
+
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.2.4.RELEASE</version>
```
Comment (ssoper-usgs) [inline pom.xml:74]: 2.2.5 is out, though I still haven't successfully gotten it past a travis build so :man_shrugging:

File: serverless.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      FUNCTION_NAME: determineRoute
+      AQTS_DATABASE_ADDRESS: ${self:custom.db.connectInfo.DATABASE_ADDRESS}
+      AQTS_DATABASE_PORT: ${self:custom.db.connectInfo.DATABASE_PORT}
+      AQTS_SCHEMA_NAME: ${self:custom.db.connectInfo.SCHEMA_NAME}
+      AQTS_SCHEMA_OWNER_USERNAME: ${self:custom.db.connectInfo.SCHEMA_OWNER_USERNAME}
+      AQTS_DATABASE_NAME: ${self:custom.db.connectInfo.DATABASE_NAME}
+      AQTS_SCHEMA_OWNER_PASSWORD: ${self:custom.db.connectInfo.SCHEMA_OWNER_PASSWORD}
+      AWS_DEPLOYMENT_REGION: ${self:provider.region}
+      ROOT_LOG_LEVEL: INFO
+      MAX_TIME_SERIES_CORRECTED_DATA: 2036854775808
```
Comment (ssoper-usgs) [inline serverless.yml]: Might not need line 42

File: src/main/java/gov/usgs/wma/waterdata/DetermineRoute.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		result.setType(determineType(request.getUniqueId()));
+		return result;
+	}
+
+	protected String determineType(String uniqueId) {
+		String type = ERROR;
+		Map<String, Object> data = dataDao.getDataTypeRouting(uniqueId);
+		LOG.debug("data: {}", data);
+		if (null != data) {
+			switch (data.get(DataDao.DATA_TYPE).toString()) {
```
Comment (ssoper-usgs) [inline src/main/java/gov/usgs/wma/waterdata/DetermineRoute.java:42]: This is neat - so if there are other data types besides `tsDailyValueStatisitcalTransform` in the future, we can just add them here?

File: src/test/resources/application.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+spring:
+  datasource:
+    url: jdbc:postgresql://${AQTS_DATABASE_ADDRESS}:${AQTS_DATABASE_PORT}/${AQTS_DATABASE_NAME}?currentSchema=${AQTS_SCHEMA_NAME}
+    username: ${AQTS_SCHEMA_OWNER_USERNAME}
+    password: ${AQTS_SCHEMA_OWNER_PASSWORD}
+    hikari:
+      maximum-pool-size: 2
+
+maxTimeSeriesCorrectedData: 10
```
Comment (ssoper-usgs) [inline src/test/resources/application.yml]: This property might be related to the MAX_TIME_SERIES_CORRECTED_DATA property in the serverless.yml file, so my previous comment about the serverless line not being needed might be false...

**Review by ssoper-usgs — APPROVED** [review]

File: src/main/java/gov/usgs/wma/waterdata/DetermineRoute.java (hunk 4 above)
Comment (dsteinich) [reply src/main/java/gov/usgs/wma/waterdata/DetermineRoute.java:42]: That's my current hope, but may get unwieldy if too may parm_cd/stat_cd combinations (or null stat_cd get too confusing)

**Pull request merged**